src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -411,8 +411,18 @@ private function readProperty(array $zval, string $property, bool $ignoreInvalid
                         throw $e;
                     }
                 } elseif (PropertyReadInfo::TYPE_PROPERTY === $type) {
-                    if ($access->canBeReference() && !isset($object->$name) && !\array_key_exists($name, (array) $object) && !(new \ReflectionProperty($class, $name))->hasType()) {
```

Comment:
Is there a reason why the `$access->canBeReference()` check has been dropped here ?

Reply:
yes: that check was just wrong - `__get` can return by reference, so this was an unrelated check

Reply:
Thanks for the information.